### PR TITLE
Match by tag index.

### PR DESCRIPTION
**Pull request by Shylock-Hg**

### CODE REVIEW

File: src/planner/match/LabelIndexSeek.h [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    std::shared_ptr<meta::cpp2::IndexItem> static selectIndex(
+        const std::shared_ptr<meta::cpp2::IndexItem> candidate,
+        const std::shared_ptr<meta::cpp2::IndexItem> income) {
+        // less fields is better
+        // TODO(shylock) rank for field itself(e.g. INT is better than STRING)
+        if (candidate->get_fields().size() > income->get_fields().size()) {
+            return income;
+        }
+        return candidate;
+    }
```
[the remark this answers is not in the document]
Comment (Shylock-Hg) [reply src/planner/match/LabelIndexSeek.h:46]: Ok, mark.

File: src/planner/match/LabelIndexSeek.cpp [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+
+#include "planner/match/LabelIndexSeek.h"
+#include "planner/Query.h"
+#include "planner/match/MatchSolver.h"
+#include "util/ExpressionUtils.h"
+
+namespace nebula {
+namespace graph {
+
+bool LabelIndexSeek::matchNode(NodeContext* nodeCtx) {
```
[the remark this answers is not in the document]
Comment (Shylock-Hg) [reply src/planner/match/LabelIndexSeek.cpp:15]: Good catch. Fixed.

**Review by CPWstatic — APPROVED** [review]

File: src/planner/match/LabelIndexSeek.h [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+
+    bool matchEdge(EdgeContext* edgeCtx) override;
+
+    StatusOr<SubPlan> transformNode(NodeContext* nodeCtx) override;
+
+    StatusOr<SubPlan> transformEdge(EdgeContext* edgeCtx) override;
+
+    static StatusOr<IndexID> pickTagIndex(const NodeContext* nodeCtx);
+
+    std::shared_ptr<meta::cpp2::IndexItem> static selectIndex(
```
Comment (yixinglu) [inline src/planner/match/LabelIndexSeek.h:37]: `static std::shared_ptr<meta::cpp2::IndexItem> selectIndex`

Comment (Shylock-Hg) [reply src/planner/match/LabelIndexSeek.h:37]: Good catch. fixed.

**Review by yixinglu — DISMISSED** [review]
LGTM

**Review by bright-starry-sky — APPROVED** [review]
LGTM

**Review by bright-starry-sky — APPROVED** [review]
LGTM

**Review by CPWstatic — APPROVED** [review]

**Pull request merged**